src/components/forms/formUtils.js

```diff
@@ -0,0 +1,46 @@
+import {FormFieldTypes} from './forms';
+import { isNil, isFunction } from 'lodash/fp';
+
+const alwaysRequiredProps = ['id'];
+const alwaysPossibleProps = [
```

Comment:
what would you think of renaming this to `commonProps` or something instead? I'm imagining like


```suggestion
const commonProps = [
    id: { type: 'string', required: true },
    name: { type: 'string' },
    description: { type: 'string' },
    disabled: { type: 'bool' }
    ...
];
```

Reply:
sure! works for me.

Reply:
do you think typechecking would be good here? i changed the name to something i think a bit better, but kept optional and required separate: `commonOptionalProps`, but i am not sure if type checking is overkill here or a good inclusion. if we want to do that i can merge all props configs like you suggested